src/main/java/serenity/TaskList.java

```diff
@@ -0,0 +1,195 @@
+package serenity;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a list of tasks, where tasks can be added, deleted
+ * and marked as done or undone.
+ */
+public class TaskList {
+
+    protected ArrayList<Task> tasks;
+
+    /**
+     * Constructs a TaskList with no existing tasks.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a TaskList with existing tasks.
+     *
+     * @param tasks ArrayList of existing tasks.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Creates task of specified type with given task description.
+     *
+     * @param input Task description specifying type of task.
+     * @return Task of specified type with given task description
+     * @throws SerenityException If type of task is not specified.
+     */
+    public Task createTask(String input) throws SerenityException {
+        Task t;
+
+        if (input.startsWith("todo")) {
+            String[] description = input.split(" ");
```

Comment:
This naming seems a bit vague. What this line is doing is creating a temporary placeholder array to check if there is anything after the command words "todo", "deadline", "event", and then the `description` array isn't used anymore, am I correct? So perhaps other names like `splitInput `or `preprocessedInput `would make the utility of the variable clearer.